eng/scripts/Language-Settings.ps1

```diff
@@ -139,15 +139,32 @@ function Get-rust-AdditionalValidationPackagesFromPackageSet ($packagesWithChang
   return $additionalPackages ?? @()
 }
 
+# $GetPackageInfoFromPackageFileFn = "Get-${Language}-PackageInfoFromPackageFile"
 function Get-rust-PackageInfoFromPackageFile([IO.FileInfo]$pkg, [string]$workingDirectory) {
-  #$pkg will be a FileInfo object for the Cargo.toml file in a package artifact directory
-  $package = cargo read-manifest --manifest-path $pkg.FullName | ConvertFrom-Json
+  # Create a temporary folder for extraction
+  $extractionPath = Join-Path ([System.IO.Path]::GetTempPath()) ([System.IO.Path]::GetRandomFileName())
```

Comment:
I strongly recommend against `Join-Path`. It's buggy, and basically just does string concat. It's funny you're using `[Path]::GetTempPath()` when you could use `$env:TEMP`, but don't use `[Path]::Combine()`, which I recommend over `Join-Path`. It actually handles both absolute and relative paths correctly. `Join-Path` will concat an absolute child path to a base path without thinking twice, which obviously is not what you want.

Reply:
I haven't seen problems with `Join-Path` but I don't mind switching.

`$env:TEMP` is not universal, on Linux you aren't guaranteed to get a value back. `[System.IO.Path]::GetTempPath()` is more universal (though it's [probably just returning `/tmp/`](https://github.com/dotnet/runtime/blob/62578463d0b13c7694cf679caa6cbb1458a35c73/src/libraries/System.Private.CoreLib/src/System/IO/Path.Unix.cs#L81)).